src/main.cpp

```diff
@@ -123,7 +140,10 @@ void main(void)
         eeprom_update();
 
         t_start += 1000;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"
         k_sleep(K_TIMEOUT_ABS_MS(t_start));
+#pragma GCC diagnostic pop
```

Comment:
This should have been fixed by https://github.com/zephyrproject-rtos/zephyr/pull/29880, which will also go into Zephyr 2.5 release.

Reply:
I believe this may be related to toolchain (gcc) not Zephyr directly.  I am building on MacOSx and unlike windows or linux, the toolchain is not provided via Zephyr.  I have to build it myself.  But I have not encountered other such warnings in the many Zephyr projects I've worked on.

Reply:
One final note on this.   My organization does not allow warning-producing code to be checked-in" all warnings must be resolved.  In our branches, I have fixed these warnings, but I pushed them to you because I thought others my have to work under the same rules as our org.

Reply:
I'm following the same rule that all compiler warnings have to be fixed. However, I'm not sure if disabling the warning is the right approach here, as warnings are usually there for a reason.

In our forked Zephyr branch this issue should be fixed also for v2.4 stable, see [here](https://github.com/LibreSolar/zephyr/commit/e272c47c6bbeccf2b2233c3b0eafc02381a6c88e). Don't think it's related to the toolchain.